File: app/handlers/echo.go
package handlers

import (
	"github.com/otiai10/hisyotan/app/routes"
	"github.com/otiai10/hisyotan/config"
	"github.com/otiai10/twistream"
	"github.com/otiai10/words"
)

// EchoHandler ...
type EchoHandler struct {
	HandlerBase
}

// Match ...
func (h EchoHandler) Match(tw twistream.Status) bool {
	if tw.InReplyToUserIdStr != config.V.Twitter.Bot.UserID {
		return false
	}
	return words.Parse(tw.Text).Has("/echo")
}

// Handle ...
func (h EchoHandler) Handle(tw twistream.Status, tl routes.Tweetable) error {
	botname := config.V.Twitter.Bot.ScreenName

	text := words.Parse(tw.Text).
		Remove("@" + botname).Remove("/echo").
		Prepend("@" + tw.User.ScreenName).Join(" ")

	return tl.Tweet(twistream.Status{
		Text:                text,
		InReplyToScreenName: tw.User.ScreenName,
		InReplyToStatusId:   tw.Id,
	})
}

File: bot/controllers/echo.rb
require 'pp'

module BOT
  class EchoController < BaseController
    def initialize
      super Regexp.new("@hisyotan -e[ 　](?<value>.+)")
    end
    def handle(status)
      user = User.where(:id_original => status.user.id).first
      return self.reply(status, "だれやねん") if user.nil?
      value = @pattern.match(status.text)["value"]
      self.reply(status, BOT.message("echo", v: value))
    end
  end
end
